### PR TITLE
Tendermint and Cosmos IPLD schemas

### PR DESCRIPTION
**Pull request by i-norden**
Opening as draft for now and will solicit review/discussion from Tendermint/Cosmos community. Some things that still need done: 

* Decide on and upstream multicodec types 
* Decide if (and how) we can more explicitly type the protobuf/amino encoded values that are stored in IAVL leafs or referenced from SMT leafs

### CODE REVIEW

**Comment by i-norden** [comment]
Work on type gen, marshallers, unmarshallers is in here: https://github.com/vulcanize/go-codec-dagcosmos

File: data-structures/cosmos/basic_types.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+# BlockIDFlag is a single byte flag
+type BlockIDFlag enum {
+  | BlockIDFlagUnknown ("0")
+  | BlockIDFlagAbsent ("1")
+  | BlockIDFlagCommit ("2")
+  | BlockIDFlagNil ("3")
+} representation int
+```
+
+## Params
```
Comment (tac0turtle) [inline data-structures/cosmos/basic_types.md]: I'm new to this repo, how would upgrades be handled? The tendermint repo has a couple changes in mind to core data structures

Comment (i-norden) [reply data-structures/cosmos/basic_types.md]: Updates to the data structures would need to be reflected in the schemas here and the implementations of the schema types and their marshallers and unmarshallers (in this repo for the Go implementation: https://github.com/vulcanize/go-codec-dagcosmos).

**Comment by i-norden** [comment]
@rvagg should I close this in favor of https://github.com/ipld/ipld/pull/111 or do we want to have these schemas in both places?

**Comment by rvagg** [comment]
@i-norden we're adding tombstones here, currently piecemeal but we'll end up doing it in bulk when we allocate some time. See https://github.com/ipld/specs/blob/master/data-structures/hashmap.md for an example.

It's up to you how you'd like to proceed with this one. If files didn't get merged then there's probably no point in leaving a tombstone and we may as well close the PR if nothing is lost in transition. For anything that got merged, it would be good if you could add tombstones for them redirecting to the most appropriate web link for that resource.

**Comment by i-norden** [comment]
Closing in favor of https://github.com/ipld/ipld/pull/111

**Pull request closed without merging**